Explorer/Assets/DCL/Backpack/BackpackEquipStatusController.cs

```diff
@@ -142,9 +157,51 @@ private async UniTaskVoid UpdateProfileAsync(CancellationToken ct)
             bool publishProfileChange = !appArgs.HasFlag(AppArgsFlags.SELF_PREVIEW_BUILDER_COLLECTIONS)
                                         && !appArgs.HasFlag(AppArgsFlags.SELF_PREVIEW_WEARABLES);
 
-            var profile = await selfProfile.UpdateProfileAsync(publish: publishProfileChange, ct);
-            MultithreadingUtility.AssertMainThread(nameof(UpdateProfileAsync), true);
-            UpdateAvatarInWorld(profile!);
+            Profile? oldProfile = await selfProfile.ProfileAsync(ct);
+
+            if (oldProfile == null)
+            {
+                ShowErrorNotificationAsync(ct).Forget();
+                return;
+            }
+
+            Profile newProfile = oldProfile.CreateNewProfileForUpdate(equippedEmotes, equippedWearables,
+                forceRender, emoteStorage, wearableStorage,
+                // Don't increment the version as it will be incremented later on selfProfile.UpdateProfileAsync
+                !publishProfileChange);
+
+            // Skip publishing the same profile
+            if (newProfile.Avatar.IsSameAvatar(oldProfile.Avatar))
+                return;
+
+            // Clone the old profile since the original will be disposed when its replaced in the cache
+            oldProfile = profileBuilder.From(oldProfile).Build();
+
+            // Update profile immediately to prevent UI inconsistencies
+            // Without this immediate update, temporary desync can occur between backpack closure and catalyst validation
+            // Example: Opening the emote wheel before catalyst validation would show outdated emote selections
+            profileCache.Set(newProfile.UserId, newProfile);
+            UpdateAvatarInWorld(newProfile);
+
+            if (!publishProfileChange) return;
+
+            try
+            {
+                var updatedProfile = await selfProfile.UpdateProfileAsync(newProfile, ct);
+                MultithreadingUtility.AssertMainThread(nameof(UpdateProfileAsync), true);
+                UpdateAvatarInWorld(updatedProfile!);
```

Comment:
I dont quite follow if you need this line. If it was successful, what are you updating here?

Reply:
Removed!